src/tests/components/AddCommentForm.test.js

```diff
@@ -0,0 +1,200 @@
+      const state = {
+        author: '',
+        commentDate: new Date(),
+        text: '',
+        emptyAuthorError: false,
+        emptyTextError: false
+      };
+      wrapper.setState(state);
+      const event = {
+        preventDefault() {}
+      };
+      wrapper.find('#addNewCommentForm').simulate('submit', event);
+      expect(wrapper.state().emptyAuthorError).toBe(true);
+      expect(wrapper.state().emptyTextError).toBe(true);
+    });
+  });
+  describe('when input is changed', () => {
+    it('should update state.author when author input is changed', () => {
+      const state = {
+        author: '',
+        commentDate: new Date(),
+        text: '',
+        emptyAuthorError: false,
+        emptyTextError: false
+      };
+      wrapper.setState(state);
+      const value = 'john';
+      const name = 'author';
+
+      wrapper
+        .find('input')
+        .at(0)
+        .simulate('change', { target: { name, value } });
+
+      expect(wrapper.state('author')).toBe('john');
+    });
+    it('should update state.text when text area is changed', () => {
+      const state = {
+        author: '',
+        commentDate: new Date(),
+        text: '',
+        emptyAuthorError: false,
+        emptyTextError: false
+      };
+      wrapper.setState(state);
+      const value = 'new comment';
+      const name = 'text';
+
+      wrapper
+        .find('textarea')
+        .at(0)
+        .simulate('change', { target: { name, value } });
+
+      expect(wrapper.state('text')).toBe('new comment');
+      const mockchangeErrorState = jest.fn();
+    });
+  });
+  describe('when changeErrorState() is called', () => {
+    it('should set emptyAuthorError to false if author is not empty', () => {
+      const state = {
+        author: 'john',
+        commentDate: new Date(),
+        text: '',
+        emptyAuthorError: false,
+        emptyTextError: false
+      };
+      wrapper.setState(state);
+      wrapper.instance().changeErrorState();
+      expect(wrapper.state('emptyAuthorError')).toBe(false);
+    });
+    it('should not set emptyAuthorError to false if author is empty', () => {
+      const state = {
+        author: '',
+        commentDate: new Date(),
+        text: '',
+        emptyAuthorError: true,
+        emptyTextError: false
+      };
+      wrapper.setState(state);
+      wrapper.instance().changeErrorState();
+      expect(wrapper.state('emptyAuthorError')).toBe(true);
+    });
+    it('should set emptyTextError to false if text is not empty', () => {
+      const state = {
+        author: '',
+        commentDate: new Date(),
+        text: 'new comment',
+        emptyAuthorError: false,
+        emptyTextError: true
+      };
+      wrapper.setState(state);
+      wrapper.instance().changeErrorState();
+      expect(wrapper.state('emptyTextError')).toBe(false);
+    });
+    it('should not set emptyTextError to false if text is empty', () => {
```

Comment:
rename these "not set X to false" into "set X to true"

Reply:
changeErrorState = () => {
    const { author, text } = this.state;
    if (author !== '') {
      this.setState({ emptyAuthorError: false });
    }
    if (text !== '') {
      this.setState({ emptyTextError: false });
    }
  };

this function doesn't set anything to true...only to false

Reply:
you are right